includes/class-sensei-frontend.php

```diff
@@ -1765,13 +1771,13 @@ public function sensei_process_registration() {
 
 		// Redirect.
 		global $wp;
-		if ( wp_get_referer() ) {
-			$redirect = esc_url( wp_get_referer() );
+		if ( ! empty( $new_user_http_referer ) ) {
+			$redirect = esc_url( $new_user_http_referer );
 		} else {
 			$redirect = esc_url( home_url( $wp->request ) );
 		}
 
-		wp_redirect( apply_filters( 'sensei_registration_redirect', $redirect ) );
+		wp_safe_redirect( apply_filters( 'sensei_registration_redirect', $redirect ) );
```

Comment:
Ideally, this is where the escaping would be done. And `esc_url_raw` is preferred over `esc_url`, as it won't decode entities.

Reply:
Done!